src/Elliptic/Systems/Xcts/Equations.hpp

```diff
@@ -0,0 +1,283 @@
+    const Scalar<DataVector>& shift_dot_deriv_extrinsic_curvature_trace,
+    const Scalar<DataVector>& conformal_factor,
+    const Scalar<DataVector>& lapse_times_conformal_factor) noexcept;
+
+/*!
+ * \brief The linearization of `add_lapse_sources`
+ *
+ * Note that this linearization is only w.r.t. \f$\psi\f$ and \f$\alpha\psi\f$.
+ * The linearization w.r.t. \f$\beta^i\f$ is added in
+ * `add_curved_linearized_momentum_sources` or
+ * `add_flat_cartesian_linearized_momentum_sources`.
+ */
+template <int ConformalMatterScale>
+void add_linearized_lapse_sources(
+    gsl::not_null<Scalar<DataVector>*> linearized_lapse_equation,
+    const Scalar<DataVector>& conformal_energy_density,
+    const Scalar<DataVector>& conformal_stress_trace,
+    const Scalar<DataVector>& extrinsic_curvature_trace,
+    const Scalar<DataVector>& dt_extrinsic_curvature_trace,
+    const Scalar<DataVector>& shift_dot_deriv_extrinsic_curvature_trace,
+    const Scalar<DataVector>& conformal_factor,
+    const Scalar<DataVector>& lapse_times_conformal_factor,
+    const Scalar<DataVector>& conformal_factor_correction,
+    const Scalar<DataVector>& lapse_times_conformal_factor_correction) noexcept;
+
+/*!
+ * \brief Add the "distortion" source term to the Hamiltonian constraint and the
+ * lapse equation.
+ *
+ * Adds \f$-\frac{1}{8}\psi^{-7}\bar{A}^{ij}\bar{A}_{ij}\f$ to the Hamiltonian
+ * constraint and \f$\frac{7}{8}\alpha\psi^{-7}\bar{A}^{ij}\bar{A}_{ij}\f$ to
+ * the lapse equation.
+ *
+ * \see Xcts
+ * \see Xcts::Tags::LongitudinalShiftMinusDtConformalMetricSquare
+ */
+void add_distortion_hamiltonian_and_lapse_sources(
+    gsl::not_null<Scalar<DataVector>*> hamiltonian_constraint,
+    gsl::not_null<Scalar<DataVector>*> lapse_equation,
+    const Scalar<DataVector>&
+        longitudinal_shift_minus_dt_conformal_metric_square,
+    const Scalar<DataVector>& conformal_factor,
+    const Scalar<DataVector>& lapse_times_conformal_factor) noexcept;
+
+/*!
+ * \brief The linearization of `add_distortion_hamiltonian_and_lapse_sources`
+ *
+ * Note that this linearization is only w.r.t. \f$\psi\f$ and \f$\alpha\psi\f$.
+ */
+void add_linearized_distortion_hamiltonian_and_lapse_sources(
+    gsl::not_null<Scalar<DataVector>*> hamiltonian_constraint,
+    gsl::not_null<Scalar<DataVector>*> lapse_equation,
+    const Scalar<DataVector>&
+        longitudinal_shift_minus_dt_conformal_metric_square,
+    const Scalar<DataVector>& conformal_factor,
+    const Scalar<DataVector>& lapse_times_conformal_factor,
+    const Scalar<DataVector>& conformal_factor_correction,
+    const Scalar<DataVector>& lapse_times_conformal_factor_correction) noexcept;
+
+/*!
+ * \brief Add the nonlinear source to the momentum constraint and add the
+ * "distortion" source term to the Hamiltonian constraint and lapse equation.
+ *
+ * Adds \f$\left((\bar{L}\beta)^{ij} -
+ * \bar{u}^{ij}\right)\left(\frac{w_j}{\alpha\psi} - 7 \frac{v_j}{\psi}\right)
```

Comment:
could you please document what `w_j` and `v_j` are?

Reply:
They are defined in the docs for the XCTS system which isn't in this PR yet, so I replaced them with \partial_j (\alpha\psi) and \partial_j \psi.